stories/ng2-component-lab/button.component.exp.ts

```diff
@@ -1,154 +1,43 @@
-        showSource: true,
-        template: `
-      <sdc-button  [color]="'negative'" >Button</sdc-button>
-    `
-    }).case('Negative Button Disabled', {
-        showSource: true,
-        template: `
-      <sdc-button  [color]="'negative'" [disabled]="true">Button</sdc-button>
-    `
-    }).case('Warning Button', {
-        showSource: true,
-        template: `
-      <sdc-button [color]="'warning'">Button</sdc-button>
-    `
-    }).case('Warning Button Disabled', {
-        showSource: true,
-        template: `
-      <sdc-button [color]="'warning'" [disabled]="true">Button</sdc-button>
-    `
-    }).case('Primary Outline', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline'">Button</sdc-button>
-    `
-    }).case('Primary Outline Disabled', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline'" [disabled]="true">Button</sdc-button>
-    `
-    }).case('Gray Outline', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline'" [color]="'gray'">Button</sdc-button>
-    `
-    }).case('Gray Outline Disabled', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline'" [color]="'gray'" [disabled]="true">Button</sdc-button>
-    `
-    }).case('Positive Outline', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline'" [color]="'positive'">Button</sdc-button>
-    `
-    }).case('Positive Outline Disabled', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline'" [color]="'positive'" [disabled]="true">Button</sdc-button>
-    `
-    }).case('Negative Outline', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline'" [color]="'negative'">Button</sdc-button>
-    `
-    }).case('Negative Outline Disabled', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline'" [color]="'negative'" [disabled]="true">Button</sdc-button>
-    `
-    })
-    .case('Primary Outline Rounded', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline-rounded'">Button</sdc-button>
-    `
-    }).case('Primary Outline Rounded Disabled', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline-rounded'" [disabled]="true">Button</sdc-button>
-    `
-    }).case('Gray Outline Rounded', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline-rounded'" [color]="'grey'">Button</sdc-button>
-    `
-    }).case('Gray Outline Rounded Disabled', {
-        showSource: true,
-        template: `
-      <sdc-button [sdcButtonStyle]="'outline-rounded'" [color]="'grey'" [disabled]="true">Button</sdc-button>
-    `
-    }).case('Positive Outline Rounded', {
+    .case('Primary Buttons : Large, Medium, Small', {
```

Comment:
Please show all the buttons types and sizes and active/disabled.

Try this code:

```
import {experimentOn} from '@islavi/ng2-component-lab';
function showButtonProps(btnType, btnSize) {
    window.alert(`Clicked!\ntype="${btnType}" size="${btnSize}"`);
};

const buttonTypes = ['primary', 'white', 'gray', 'positive', 'negative', 'warning', 'secondary'];
const buttonSizes = ['large', 'medium', 'small'];
const experiment = experimentOn('Button');
buttonTypes.forEach((buttonType) => {
    [false, true].forEach((buttonDisabled) => {
        experiment.case(`Button ${buttonType}${buttonDisabled ? ' disabled' : ''}`, {
            showSource: true,
            context: {
                showButtonProps
            },
            template: buttonSizes.map((buttonSize) =>
                `<sdc-button text="${buttonType}" type="${buttonType}" size="${buttonSize}" (click)="showButtonProps('${buttonType}', '${buttonSize}')" ${buttonDisabled ? ' [disabled]="true"' : ''}></sdc-button>`).join('\n')
        });
    });
});
export default experiment;
```

Reply:
👍